### ISSUE TITLE
Cut a release

### ISSUE THREAD

**Issue by mrkkrp** [commercialhaskell/path#142]
Last release was in 2017, come on. There are lots of changes, such as `Lift` instance I needed just today.

**Comment by mrkkrp** [comment]
Also I propose to cut support for all GHC versions but 3 most recent ones.

**Comment by mrkkrp** [comment]
@NorfairKing Could you update the `stack.yaml` file to use one of the newer resolvers and update `genvalidity` dependencies? I attempted that myself, but there are compilation errors.

**Comment by NorfairKing** [comment]
```
diff --git a/stack.yaml b/stack.yaml
index 6635304..817a7b2 100644
--- a/stack.yaml
+++ b/stack.yaml
@@ -1,5 +1,2 @@
 extra-deps:
-- genvalidity-0.3.2.0
-- genvalidity-property-0.0.0.0
-- validity-0.3.3.0
-resolver: lts-8.3
+resolver: lts-14.0
```

**Comment by NorfairKing** [comment]
@mrkkrp I just pushed to `upgrade-deps`.
By upgrading the dependencies, I found these bugs:

```
  1) Parsing: Path Abs Dir Produces valid paths when it succeeds
       Falsifiable (after 34 tests and 3 shrinks):
         "/a.."
       predicate failed on: "/a../"

  To rerun use: --match "/Parsing: Path Abs Dir/Produces valid paths when it succeeds/"

  src/Test/Validity/Functions/CanFail.hs:98:25: 
  2) Parsing: Path Rel Dir Produces valid paths when it succeeds
       Falsifiable (after 31 tests and 4 shrinks):
         "..a"
       predicate failed on: "..a/"

  To rerun use: --match "/Parsing: Path Rel Dir/Produces valid paths when it succeeds/"

  src/Test/Validity/Functions/CanFail.hs:98:25: 
  3) Parsing: Path Abs File Produces valid paths when it succeeds
       Falsifiable (after 34 tests and 3 shrinks):
         "/a.."
       predicate failed on: "/a.."

  To rerun use: --match "/Parsing: Path Abs File/Produces valid paths when it succeeds/"

  src/Test/Validity/Functions/CanFail.hs:98:25: 
  4) Parsing: Path Rel File Produces valid paths when it succeeds
       Falsifiable (after 31 tests and 4 shrinks):
         "..a"
       predicate failed on: "..a"
```

**Comment by mrkkrp** [comment]
Interesting. Now someone probably should fix them.

**Comment by NorfairKing** [comment]
I'm not sure who to ping about this
I need someone else to verify the validity constraints that I've written.

**Comment by mrkkrp** [comment]
I don't have much time to spend on this, but eventually I'll get to this if no one else can.

**Comment by NorfairKing** [comment]
It seems that the stack build is also broken on master.

**Comment by mrkkrp** [comment]
I’m on vacations. If you fix the build we can release though.

**Comment by NorfairKing** [comment]
@mrkkrp It seems that the failures were false positives. I think I managed to fix them correctly.

**Comment by mrkkrp** [comment]
I’ll take a look in the evening.

**Comment by NorfairKing** [comment]
@mrkkrp Is there anything else you need?

**Comment by mrkkrp** [comment]
No, but am I needed at all to cut a release? In any case, I can do it later today, or you can do it yourself.

**Comment by NorfairKing** [comment]
I don't know if I am able to, permissions wise. Feel free to go ahead. I think the code is ready for release.

**Comment by mrkkrp** [comment]
Tagged and released 0.7.0 on Hackage.
